File: CartyObjC/CartyObjC/BannerViewController.m
//
//  BannerViewController.m
//  CartyObjC
//
//  Created by GZTD-03-01959 on 2026/1/5.
//

#import "BannerViewController.h"
#import <CartySDK/CartySDK.h>

@interface BannerViewController ()<CTBannerAdDelegate>
@property (nonatomic,weak)IBOutlet UIView *adView;
@property (nonatomic,weak)IBOutlet UILabel *infoLabel;
@property (nonatomic,strong)CTBannerAd *bannerAd;
@end

@implementation BannerViewController

- (void)viewIsAppearing:(BOOL)animated
{
    [super viewIsAppearing:animated];
    [self.adView addSubview:self.bannerAd];
    self.bannerAd.center = CGPointMake(self.adView.bounds.size.width/2.0, self.adView.bounds.size.height/2.0);
}

- (void)viewDidLoad
{
    [super viewDidLoad];
    self.bannerAd = [[CTBannerAd alloc] init];
    self.bannerAd.placementid = @"281217152640";
    self.bannerAd.rootViewController = self;
    self.bannerAd.bannerSize = CTBannerSizeType320x50;
    self.bannerAd.delegate = self;
}


-(IBAction)loadAd:(id)sender
{
    self.infoLabel.text = @"Start load Ad";
    [self.bannerAd loadAd];
}

- (void)CTBannerAdDidLoad:(nonnull CTBannerAd *)ad { 
    self.infoLabel.text = @"Ad loaded";
}

- (void)CTBannerAdDidShow:(nonnull CTBannerAd *)ad { 
    self.infoLabel.text= @"Ad did show";
}

- (void)CTBannerAdLoadFail:(nonnull CTBannerAd *)ad withError:(nonnull NSError *)error { 
    self.infoLabel.text = [NSString stringWithFormat:@"Ad load fail %@",error];
}

- (void)CTBannerAdShowFail:(nonnull CTBannerAd *)ad withError:(nonnull NSError *)error { 
    self.infoLabel.text = [NSString stringWithFormat:@"Ad show fail %@",error];
}

- (void)CTBannerAdDidClick:(nonnull CTBannerAd *)ad {
    self.infoLabel.text = @"Ad click";
}

- (void)CTBannerAdDidClose:(nonnull CTBannerAd *)ad {
    self.infoLabel.text = @"Ad close";
}
@end

File: CartySwift/CartySwift/BannerViewController.swift
//
//  BannerViewController.swift
//  CartySwift
//
//  Created by GZTD-03-01959 on 2026/1/5.
//

import UIKit
import CartySDK

class BannerViewController: UIViewController,CTBannerAdDelegate
{
    @IBOutlet var infoLabel:UILabel?
    @IBOutlet var adView:UIView?
    let bannerAd = CTBannerAd()
    
    override func viewIsAppearing(_ animated: Bool) {
        super.viewIsAppearing(animated)
        if(adView != nil)
        {
            adView!.addSubview(bannerAd)
            bannerAd.center = CGPointMake(adView!.bounds.size.width/2.0, adView!.bounds.size.height/2.0)
        }
    }
    override func viewDidLoad()
    {
        super.viewDidLoad()
        bannerAd.placementid = "281217152640"
        bannerAd.rootViewController = self
        bannerAd.bannerSize = CTBannerSizeType320x50
        bannerAd.delegate = self
    }
    
    @IBAction func loadAd()
    {
        infoLabel?.text = "Start load Ad"
        bannerAd.load()
    }
    
    func ctBannerAdDidLoad(_ ad: CTBannerAd) {
        infoLabel?.text = "Ad loaded"
    }
    
    func ctBannerAdLoadFail(_ ad: CTBannerAd, withError error: any Error) {
        infoLabel?.text = "Ad load fail \(error)"
    }
    
    func ctBannerAdDidShow(_ ad: CTBannerAd) {
        infoLabel?.text = "Ad did show"
    }
    
    func ctBannerAdShowFail(_ ad: CTBannerAd, withError error: any Error) {
        infoLabel?.text = "Ad show fail \(error)"
    }
    
    func ctBannerAdDidClick(_ ad: CTBannerAd) {
        infoLabel?.text = "Ad click"
    }
    
    func ctBannerAdDidClose(_ ad: CTBannerAd) {
        infoLabel?.text = "Ad close"
    }
}
